samcli/commands/deploy/utils.py

```diff
@@ -89,3 +87,13 @@ def sanitize_parameter_overrides(parameter_overrides):
     :return:
     """
     return {key: value.get("Value") if isinstance(value, dict) else value for key, value in parameter_overrides.items()}
+
+
+def hide_noecho_parameter_overrides(template_parameters, parameter_overrides):
+    hidden_params = copy.deepcopy(parameter_overrides)
+    params = template_parameters.get("Parameters", None)
+    for key, value in hidden_params.items():
+        if isinstance(params, dict) and key in params and isinstance(params[key], dict):
```

Comment:
Is it better to:
1. Move `isinstance(params, dict)` out of the loop, something like 
```
if not isinstance(params, dict):
  return hidden_params
for ...
```
2. better to remove the check `key in params` and use `params.get(key)` instead of `params[key]`?